src/tests/ftest/util/environment_utils.py

```diff
@@ -0,0 +1,560 @@
+    def user_dir(self, value):
+        """Set the user directory path.
+
+        Args:
+            value (str): the user directory path
+        """
+        self.__set_value('user_dir', value)
+
+    def default_user_dir(self):
+        """Get the default user directory path.
+
+        Returns:
+            str: the default user directory path
+        """
+        return os.path.join(self.log_dir, "user")
+
+    @property
+    def interface(self):
+        """Get the interface device.
+
+        Returns:
+            str: the interface device
+        """
+        return os.environ.get(self.__ENV_VAR_MAP['interface'])
+
+    @interface.setter
+    def interface(self, value):
+        """Set the interface device.
+
+        Args:
+            value (str): the interface device
+        """
+        self.__set_value('interface', value)
+
+    def default_interface(self, logger, hosts):
+        """Get the default interface.
+
+        Args:
+            logger (Logger): logger for the messages produced by this method
+            hosts (NodeSet): hosts on which to find a homogeneous interface
+
+        Raises:
+            TestEnvironmentException: if there is a problem finding active network interfaces when
+                hosts are provided
+
+        Returns:
+            str: the default interface; can be None
+        """
+        interface = os.environ.get("OFI_INTERFACE")
+        if interface is None and hosts:
+            # Find all the /sys/class/net interfaces on the launch node (excluding lo)
+            logger.debug("Detecting network devices - OFI_INTERFACE not set")
+            try:
+                interface = get_fastest_interface(logger, hosts)
+            except NetworkException as error:
+                raise TestEnvironmentException("Error obtaining a default interface!") from error
+        return interface
+
+    @property
+    def provider(self):
+        """Get the provider.
+
+        Returns:
+            str: the provider
+        """
+        return os.environ.get(self.__ENV_VAR_MAP['provider'])
+
+    @provider.setter
+    def provider(self, value):
+        """Set the provider.
+
+        Args:
+            value (str): the provider or provider alias
+        """
+        alias = PROVIDER_ALIAS.get(value, None)
+        if alias is not None:
+            self.__set_value('provider', alias)
+        else:
+            self.__set_value('provider', value)
+
+    def default_provider(self, logger, hosts):
+        """Get the default provider.
+
+        Args:
+            logger (Logger): logger for the messages produced by this method
+            hosts (NodeSet): hosts on which to find a homogeneous provider
+
+        Raises:
+            TestEnvironmentException: if there is an error obtaining the default provider when hosts
+                are provided
+
+        Returns:
+            str: the default provider; can be None
+        """
+        if not hosts:
+            return None
```

Comment:
It seems it can only be None if the caller didn't pass hosts. Shouldn't the caller just not call this?